front/app/containers/ProjectsShowPage/shared/pb/PBExpenses.tsx

```diff
@@ -233,69 +234,66 @@ const SubmitExpensesButton = styled(Button)<{ viewMode: 'row' | 'column' }>`
   `}
 `;
 
-interface InputProps {
+interface Props {
   participationContextId: string | null;
   participationContextType: IParticipationContextType;
   viewMode: 'row' | 'column';
   className?: string;
 }
 
-interface DataProps {
-  locale: GetLocaleChildProps;
-  tenant: GetAppConfigurationChildProps;
-  basket: GetBasketChildProps;
-  project: GetProjectChildProps;
-  phase: GetPhaseChildProps;
-}
-
-interface Props extends InputProps, DataProps {}
-
-interface State {
-  processing: boolean;
-}
-
-class PBExpenses extends PureComponent<Props & InjectedIntlProps, State> {
-  constructor(props) {
-    super(props);
-    this.state = {
-      processing: false,
-    };
-  }
-
-  handleSubmitExpensesOnClick = async () => {
-    const { basket } = this.props;
+const PBExpenses = memo(
+  ({
+    participationContextType,
+    participationContextId,
+    className,
+    viewMode,
+    intl: { formatMessage },
+  }: Props & InjectedIntlProps) => {
+    const [processing, setProcessing] = useState(false);
+    const locale = useLocale();
+    const appConfiguration = useAppConfiguration();
+    const project = useProject({
+      projectId:
+        participationContextType === 'project' ? participationContextId : null,
+    });
+    const phase = usePhase(
+      participationContextType === 'phase' ? participationContextId : null
+    );
+    function getBasketId() {
+      let basketId: string | null = null;
+
+      if (participationContextType === 'project') {
+        basketId = !isNilOrError(project)
+          ? project.relationships.user_basket?.data?.id || null
+          : null;
+      } else {
+        basketId = !isNilOrError(phase)
+          ? phase.relationships.user_basket?.data?.id || null
+          : null;
+      }
 
-    if (!isNilOrError(basket)) {
-      const now = moment().format();
-      this.setState({ processing: true });
-      await updateBasket(basket.id, { submitted_at: now });
-      trackEventByName(tracks.basketSubmitted);
-      this.setState({ processing: false });
+      return basketId;
     }
-  };
-
-  render() {
-    const {
-      locale,
-      tenant,
-      participationContextType,
-      participationContextId,
-      project,
-      phase,
-      basket,
-      className,
-      viewMode,
-      intl: { formatMessage },
-    } = this.props;
-    const { processing } = this.state;
+    const basketId = getBasketId();
+    const basket = useBasket(basketId);
+
+    const handleSubmitExpensesOnClick = async () => {
+      if (!isNilOrError(basket)) {
+        const now = moment().format();
+        setProcessing(true);
+        await updateBasket(basket.id, { submitted_at: now });
```

Comment:
There should be one indeed. I tried adding one, and then came to the conclusion that there's no straightforward place to put an error in the current UI (the horizontal mode). 

Then I also noticed we're not dealing with errors anwhere updateBasket is used. So given the time constraint and that the problem is bigger than initially expected, I suggest I add it to this epic's debt list (which will be logged and we can tackle properly).

Reply:
We are doing error handling badly but you can at least add a try catch so that the screen doesn't crash in case of an error. Like: 
```
try {
updateBasket
} catch {
// Do nothing
}
```

Reply:
Will add